awx/main/models/events.py

```diff
@@ -520,13 +520,22 @@ def _update_host_summary_from_stats(self, hostnames):
                 (summary['host_id'], summary['id'])
                 for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id')
             )
+            updated_hosts = set()
             for h in all_hosts:
                 # if the hostname *shows up* in the playbook_on_stats event
                 if h.name in hostnames:
                     h.last_job_id = job.id
+                    updated_hosts.add(h)
                 if h.id in host_mapping:
                     h.last_job_host_summary_id = host_mapping[h.id]
-            Host.objects.bulk_update(all_hosts, ['last_job_id', 'last_job_host_summary_id'])
```

Comment:
One notable change is that the prior version (here) was updating `all_hosts`, even though many of those didn't actually _change_.  The new version tracks which hosts are actually *part* of this run, and only updates those.  I expect that this explains the behavior described in https://github.com/ansible/awx/issues/8145; in a scenario where you've got many parallel job slices all finishing around the same time, they're all competing with each other to update all of the same rows (even though each slice only represents some percentage of the total inventory).